packages/graphql-server-micro/src/microApollo.ts

```diff
@@ -0,0 +1,75 @@
+import { GraphQLOptions, HttpQueryError, runHttpQuery } from 'graphql-server-core';
+import * as GraphiQL from 'graphql-server-module-graphiql';
+import { json } from 'micro';
+import * as url from 'url';
+import {IncomingMessage, ServerResponse} from 'http';
+
+export interface MicroGraphQLOptionsFunction {
+  (req?: IncomingMessage): GraphQLOptions | Promise<GraphQLOptions>;
+}
+
+export function microGraphql(options: GraphQLOptions | MicroGraphQLOptionsFunction) {
+  if (!options) {
+    throw new Error('Apollo Server requires options.');
+  }
+
+  if (arguments.length > 1) {
+    throw new Error(`Apollo Server expects exactly one argument, got ${arguments.length}`);
+  }
+
+  return async function (req: IncomingMessage, res: ServerResponse) {
+    let query;
+    if (req.method === 'POST') {
+      try {
+        query = await json(req);
```

Comment:
json is a function supplied with micro that is an async parser.  the reason it is async is because it will use the request as a stream and will return the complete json once the full request payload is received.